extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/service/TransferProcessServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.service;
+
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.command.CancelTransferCommand;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.command.DeprovisionRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+import static java.util.stream.Collectors.toList;
+
+public class TransferProcessServiceImpl implements TransferProcessService {
+    private final TransferProcessStore transferProcessStore;
+    private final TransferProcessManager manager;
+    private final TransactionContext transactionContext;
+
+    public TransferProcessServiceImpl(TransferProcessStore transferProcessStore, TransferProcessManager manager, TransactionContext transactionContext) {
+        this.transferProcessStore = transferProcessStore;
+        this.manager = manager;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public @Nullable TransferProcess findById(String transferProcessId) {
+        return transactionContext.execute(() -> transferProcessStore.find(transferProcessId));
+    }
+
+    @Override
+    public @NotNull Collection<TransferProcess> query(QuerySpec query) {
+        return transactionContext.execute(() -> transferProcessStore.findAll(query).collect(toList()));
+    }
+
+    @Override
+    public @Nullable String getState(String transferProcessId) {
```

Comment:
Is it a known decision not to use Java's `Optional` instead of nullable values?

Reply:
That is the existing codebase, we can raise in dev channel